app/src/main/java/com/example/dogbook/main/fragments/MapFragment.java

```diff
@@ -104,7 +106,10 @@ private void loadedMap(GoogleMap googleMap) {
             @Override
             public void onCameraIdle() {
                 visibleAreaBox = map.getProjection().getVisibleRegion().latLngBounds;
```

Comment:
Maybe you should pass this in as a parameter to the methods?  Do you actually need it as an instance variable?